File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.java
class Solution {
    public String solution(String[] my_strings, int[][] parts) {
        String answer = "";
        
        for(int i=0;i<my_strings.length;i++){
            String p = my_strings[i];
            
            answer += p.substring(parts[i][0],parts[i][1]+1);
        }
        return answer;
    }
}

File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* my_strings[], size_t my_strings_len, int** parts, size_t parts_rows, size_t parts_cols) {
    char* answer = (char*)malloc(my_strings_len * 100);
    int len = 0;
    
    for(int i=0;i<my_strings_len;i++){
        for(int j=parts[i][0];j<=parts[i][1];j++){
            answer[len++] = my_strings[i][j];
        }
    }
    answer[len] = '\0';
    return answer;
}